### PR TITLE
Add Visual Mode

### PR DESCRIPTION
**Pull request by lovetocode999**
Edit: This PR also adds deletion with the 'x' key.

### CODE REVIEW

**Comment by matthewsot** [comment]
Thanks for the PR and many improvements, I'll try to take a look tomorrow!

File: README.md
Comment (matthewsot) [inline README.md]: Perhaps we should also add the `"currentSequence": ""` line here? Or better yet, to prevent from having to keep this updated, maybe we can just replace this with:
```javascript
vim = {
    ...
    "keys": {
        "move": "hjkl",
        "escapeSeq": "jk",
    }
};
```
and have the user pattern-match the "..." themselves.

File: docs-vim.js
Comment (matthewsot) [inline docs-vim.js]: Nitpick: Not completely related to this PR, but I think this could benefit from a few short comments describing each of the elements in this dictionary/object. At least `num`, on a first read it's not clear what we'll be using it for. Maybe something along the lines of "User-entered number buffer for the number of times to repeat a key press"? (I could be misunderstanding its use)

File: docs-vim.js
Comment (matthewsot) [inline docs-vim.js:34]: Nit: Is it OK to reset `vim.num` here, and `vim.currentSequence` in `vim.switchToNormalMode`? We might just want to have a generic "setMode" which resets these and sets the desired mode. We can handle this in a different issue/PR though.

File: docs-vim.js
Comment (matthewsot) [inline docs-vim.js]: Nit: for consistency, can we add +2 spaces here?

File: docs-vim.js
Comment (matthewsot) [inline docs-vim.js]: Nit: for consistency, can we use spaces here?

File: docs-vim.js
Comment (matthewsot) [reply docs-vim.js]: Or actually, I think just spaces instead of tabs would fix it.

File: docs-vim.js
Comment (matthewsot) [inline docs-vim.js]: I think this is duplicated with `normal_keydown`. Not super important to resolve in this PR, but we can do a code-deduplication pass at some point in the future and see if we can refactor stuff like this out.

File: docs-vim.js
Comment (matthewsot) [inline docs-vim.js]: There seem to be a bunch of `\t` tabs instead of spaces, can we switch them over for consistency?

**Review by matthewsot — CHANGES_REQUESTED** [review]
It looks really great to me, thanks! And sorry again for taking so long to get to the PR.

I made a couple of comments, but most of them can wait for future issues/PRs/etc. I think we can merge once:
1. Make sure the documentation example about using QWERTY lines up with the code, or add `...` instead so we don't have to keep updating it.
2. Tabs -> spaces everywhere.
3. Short comment explaining at least `vim.num`.

If you could take a crack at removing some of the duplicated code, that would also be great. I think what would be ideal maybe is just having something like a shared:
```javascript
vim.addKeyMappings = function (baseMap) {
    baseMap[vim.keys.move[0]] = "ArrowLeft";
    baseMap[vim.keys.move[0]] = "ArrowDown";
    baseMap[vim.keys.move[0]] = "ArrowUp";
    baseMap[vim.keys.move[0]] = "ArrowRight";
    return baseMap;
}
```
Then, the other code could just be like
```javascript
var keyMap = vim.addKeyMappings({ "Backspace": "ArrowLeft", "x": "Delete" });
```

If you're busy, I can also go ahead and make the changes. Otherwise feel free to re-request a review once 1--3 are done and it should be good to merge!

File: docs-vim.js
Comment (matthewsot) [inline docs-vim.js]: Nit: Semicolon at the end of this line.

File: docs-vim.js
Comment (matthewsot) [inline docs-vim.js]: Nit: Can we add a semicolon at the end of this line for consistency?

**Review by matthewsot — APPROVED** [review]
Looks great! I noted two lines missing semicolons, once those are added feel free to merge.

**Comment by lovetocode999** [comment]
Thanks! I added the missing semicolons, but I can't merge because I don't have write access.

**Comment by matthewsot** [comment]
No worries, merged it! Thanks for putting this together.

**Pull request merged**
